lib/resque_bus/adapter.rb

```diff
@@ -29,12 +29,15 @@ def queuebus_redis
 
             # Save the default Resque redis connection
             default_redis = ::Resque.redis
-            # Get Resque to generate the connection given the server definition
-            ::Resque.redis = @queuebus_redis_url
-            # Store the Redis namespace that Resque created...
-            @queuebus_redis = ::Resque.redis
-            # ...and restore the default.
-            ::Resque.redis = default_redis
+            begin
+              # Get Resque to generate the connection given the server definition
+              ::Resque.redis = @queuebus_redis_url
+              # Store the Redis namespace that Resque created...
+              @queuebus_redis = ::Resque.redis
+            ensure
```

Comment:
it seems like this would be a good case to TDD, to see if we can find a case that breaks this (and thus requires the ensure)